### PR TITLE
[WASM] Implement timers.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by kumpera** [comment]
Fixes https://github.com/mono/mono/issues/6369

File: mcs/class/corlib/System.Threading/Timer.cs [hunk 1]
```diff
@@ -33,8 +33,35 @@
...
+		static Dictionary<int, Action> callbacks;
+		static int next_id;
+
+		[MethodImplAttribute(MethodImplOptions.InternalCall)]
+		static extern void SetTimeout (int timeout, int id);
+
+		internal static void ScheduleTimeout (int timeout, Action action) {
+			if (callbacks == null)
+				callbacks = new Dictionary<int, Action> ();
+			int id = ++next_id;
```
Comment (marek-safar) [inline mcs/class/corlib/System.Threading/Timer.cs]: Shouldn't this be atomic?

Comment (luhenry) [reply mcs/class/corlib/System.Threading/Timer.cs]: There is no concurrency in WASM

**Comment by kumpera** [comment]
@monojenkins build Linux AArch64

File: mcs/class/corlib/System.Threading/Timer.cs (hunk 1 above)
Comment (kumpera) [reply mcs/class/corlib/System.Threading/Timer.cs]: Yes, WebAssembly is single threaded for now. There is work to make it multi-threaded, but that's years away from having browsers shipping.

**Comment by kumpera** [comment]
@monojenkins build OS X x64

**Comment by kumpera** [comment]
@monojenkins build Linux x64

**Comment by kumpera** [comment]
@monojenkins build Linux x64 FullAOT

**Pull request merged**